LibertyDSNP/example-client
# Feat/record and display handles

Purpose
---------------
We need to show handle information in the example client. Unfortunately, this information isn't available from the profiles we are already retrieving, so it must be derived from DSNP registry update events.

Solution
---------------
This PR adds a subscription for DSNP registry events. The event data is used to associate users' handles with their profile data. This information is stored in the profile slice of the redux store. We use it to show the handle in the current user's profile and in the posts feed if the sender's name is not available.

Change summary
---------------
* Update to latest SDK (to get registry update subscription feature)
* Modify populate script to add handles to accounts and register them (note there are some fixes here to solve some concurrency issues).
* Change user slice to store DSNP id rather than profile. Th profile is then associated with a profile from profile slice. At login, the current user DSNP id is currently hard coded to one of the id's generated in the populate script to demonstrate this association. This will be replaced by an address to DSNP id lookup in a future story.
* Show handle as well as name in PostList.
* Show handle in current user Profile.


Steps to Verify
----------------
1. Start with a clean chain with contracts deployed.
2. Run populate script `node scripts/populate.mjs`
3. Start example-client.
4. Log in with metamask.
5. Note that Profile now has name, handle and social address (DSNP id) of one of the accounts in the populate script (probably "An Bai").
6. Click on "All Posts".
7. Note that all feed items have a name and handle.
